diff --git a/include/verilated_types.h b/include/verilated_types.h
index cbd05ad96..f4c7ef674 100644
--- a/include/verilated_types.h
+++ b/include/verilated_types.h
@@ -1061,8 +1061,8 @@ public:
     // METHODS
 public:
     // Raw access
-    WData* data() { return &m_storage[0]; }
-    const WData* data() const { return &m_storage[0]; }
+    WData* data() { return (WData*)&m_storage[0]; }
+    const WData* data() const { return (const WData*)&m_storage[0]; }
 
     T_Value& operator[](size_t index) { return m_storage[index]; }
     const T_Value& operator[](size_t index) const { return m_storage[index]; }